apps/builder/app/shared/canvas-api.ts

```diff
@@ -0,0 +1,73 @@
+import { setInert, resetInert } from "../canvas/shared/inert";
+
+const apiWindowNamespace = "__webstudio__$__canvasApi";
+
+const _canvasApi = {
+  setInert,
+  resetInert,
+};
+
+declare global {
+  interface Window {
+    [apiWindowNamespace]: typeof _canvasApi;
+  }
+}
+
+const isInIframe = () => {
+  try {
+    return window.self !== window.top;
+  } catch (e) {
+    return true;
+  }
+};
+
+const getIframeApi = () => {
+  if (isInIframe()) {
+    // Inside the iframe, use the local window.api
+    return _canvasApi;
+  } else {
```

Comment:
after return else is not really needed, confuses me, though In know some people find it more readable